### DISCUSSION TITLE
What am I supposed to do if I remove a column from a table? How to handle this with the migrator?

### DISCUSSION THREAD

**Discussion by madsane29** [simolus3/drift#2026] [Q&A] [answered]
Did not find anything like m.removeColumn()...

Or what am I supposed to do if the given column is no longer needed in the given table?

Would the app crash if I simply remove from the entity class?

```
class MyEntity extends Table {
  IntColumn? get id => integer().autoIncrement()();

  IntColumn? get secSecId=> integer().nullable()(); <-- I don't need anymore this one
}
```

**Comment by simolus3** [comment, marked as answer]
> Would the app crash if I simply remove from the entity class?

If the column has been nullable before, I don't think this is particularly risky. Drift won't crash when reading a row with additional columns from the database. And since the column is nullable, there shouldn't be any issues inserting the column either.

That said, you can run a [simple migration](https://drift.simonbinder.eu/docs/advanced-features/migrations/#deleting-columns) to delete the column from the database as well.
